python/cudf/cudf/api/extensions/accessor.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+from cudf.utils.docutils import docfmt_partial
+import warnings
+
+import cudf
+from pandas.core.accessor import CachedAccessor
+
+
+_docstring_register_accessor = """
+    Extends `cudf.{klass}` with custom defined accessor
+
+    Parameters
+    ----------
+    name : str
+        The name to be registered in `{klass}` for the custom accessor
+
+    Returns
+    -------
+    decorator : callable
+        Decorator function for accessor
+
+    Notes
+    -----
+    The `{klass}` object will be passed to your custom accessor upon first
+    invocation. And will be cached for future calls.
+
+    If the data passed to your accessor is of wrong datatype, you should
+    raise an `AttributeError` in consistent with other cudf methods.
+
+
+    Examples
+    --------
+    {example}
+"""
+
+_dataframe_example = """
+    In your library code:
+
+        .. code-block:: python
```

Comment:
So far we've avoided using reST in our docstrings/examples. We've preferred [doctest](https://docs.python.org/3/library/doctest.html)-like examples, i.e.,

```
Examples
----------

In your library code:

>>> import cudf as gd
>>> @gd.api.extensions.register_dataframe_accessor("point")
... class PointsAccessor:
...     def __init__(self, obj):
...         self._validate(obj)
...         self._obj = obj
...     @staticmethod
...     def _validate(obj):
...         cols = obj.columns
...         if not all([vertex in cols for vertex in ["x", "y"]]):
...             raise AttributeError("Must have vertices 'x', 'y'.")
...     @property
...     def bounding_box(self):
...         xs, ys = self._obj["x"], self._obj["y"]
...         min_x, min_y = xs.min(), ys.min()
...         max_x, max_y = xs.max(), ys.max()
...         return (min_x, min_y, max_x, max_y)
```

Obviously, that's much more suitable for interactive examples, which this is not.

I'm OK with using reST -- it's just a bit painful to write. I also don't know how they end up rendering in the rapids.ai docs.

Thoughts?

Reply:
What does reST stands for?

Reply:
This is how it looks like. I notice for doctest style you can toggle between showing/hiding the outputs in the doc. For code-blocks you can't (even for IPython blocks).

![image](https://user-images.githubusercontent.com/13521008/94197511-c0089d80-fe6a-11ea-8fa1-854c86a8fdb4.png)

Reply:
Sorry -- by `reST`, I meant [ReStructuredText](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html) which is the markup format being used here.

I think the rendered docs look great! Just a question of using reST v/s the much simpler docutils style. @kkraus14 any opinions?

Reply:
doctest style is preferred

Reply:
For the "Library Code" section alone though, I kinda feel like `reST` has an edge of advantage. Most noticeably the dots on the side and lack of empty rows reduce the readability a bit. Doctest style renders:
![image](https://user-images.githubusercontent.com/13521008/94204752-b4bb6f00-fe76-11ea-921f-111f12191194.png)
What do you think?

Reply:
I agree. I don't like that you can't collapse it, but given that this isn't an interactive example, doctest seems to be a bit less appropriate.

Reply:
In general we've consistently used doctest throughout the entire library (sans one place in groupby that we should fix), so my thought is we should just keep consistency and not mix and match formats.